libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -95,61 +94,129 @@ impl<'a> Validator<'a> {
 
-                        diagnostics.push_error(error);
+                            diagnostics.push_error(error);
+                        }
                     }
                 }
-            }
 
-            let mut new_errors = self.validate_relation_arguments_bla(
-                schema,
-                ast_schema.find_model(&model.name).expect(STATE_ERROR),
-                model,
-                &constraint_names,
-            );
+                for field in model.relation_fields() {
+                    let ast_field = ast_model
+                        .fields
+                        .iter()
+                        .find(|ast_field| ast_field.name.name == field.name);
 
-            diagnostics.append(&mut new_errors);
+                    let field_span = ast_field.map(|f| f.span).unwrap_or_else(ast::Span::empty);
+
+                    if let Some(fk_name) = field.relation_info.fk_name.as_ref() {
+                        if let Some(scope) = namespace_violation_scope(fk_name, "fk") {
+                            let span = ast_field
+                                .and_then(|f| f.span_for_argument("relation", "map"))
+                                .unwrap_or(field_span);
+
+                            let message = format!(
+                            "The given constraint name `{}` has to be unique in the following namespace: {}. Please provide a different name using the `map` argument.",
+                            fk_name, scope
+                        );
+
+                            let error =
+                                DatamodelError::new_attribute_validation_error(&message, RELATION_ATTRIBUTE_NAME, span);
+                            diagnostics.push_error(error);
+                        }
+                    }
+                }
+
+                for index in &model.indices {
+                    if let Some(idx_name) = &index.db_name {
+                        if let Some(scope) = namespace_violation_scope(idx_name, "idx") {
+                            let span = ast_model.span;
+                            let message = format!(
+                            "The given constraint name `{}` has to be unique in the following namespace: {}. Please provide a different name using the `map` argument.",
+                            idx_name, scope
+                        );
+
+                            let error = DatamodelError::new_attribute_validation_error(
+                                &message,
+                                if index.is_unique() {
+                                    UNIQUE_ATTRIBUTE_NAME
+                                } else {
+                                    INDEX_ATTRIBUTE_NAME
+                                },
+                                span,
+                            );
+                            diagnostics.push_error(error);
+                        }
+                    }
+                }
+            }
         }
+
+        diagnostics
     }
```

Comment:
I find the nesting of loops and conditionals makes that method hard to read. It's the general style in this file though, that can be addressed later. One quick win might be to split it into multiple methods.